client/src/pages/board/KanbanBoard.jsx

```diff
@@ -0,0 +1,297 @@
+      tasks: {},
+      columns: {},
+      columnOrder: [],
+    }
+    await data.columns.map(column => {
+      newData.columns[column._id] = {
+        id: column._id,
+        title: column.name,
+        taskIds: column.cards.map(card => card._id)
+      }
+      column.cards.map(card => {
+        newData.tasks[card._id] = {
+          id: card._id,
+          content: card.name
+        }
+      })
+      newData.columnOrder.push(column._id)
+    })
+    setData(newData)
+  }
+
+  useEffect(() => {
+    if (update) {
+      convertAPIData()
+      setUpdate(false)
+    }
+  }, [update])
+
+  const classes = useStyles();
+  const onDragEnd = async (result) => {
+    const { destination, source, draggableId, type } = result;
+
+    if (!destination) {
+      return;
+    }
+
+    if (
+      destination.droppableId === source.droppableId &&
+      destination.index === source.index
+    ) {
+      return;
+    }
+
+    if (type === 'column') {
+      const newColumnOrder = Array.from(data.columnOrder);
+      newColumnOrder.splice(source.index, 1);
+      newColumnOrder.splice(destination.index, 0, draggableId);
+
+      const newState = {
+        ...data,
+        columnOrder: newColumnOrder,
+      };
+
+      await setData(newState);
+      await saveBoard(data.id, {columns: newState.columnOrder});      
+      return;
+    }
+
+    const start = data.columns[source.droppableId];
+    const finish = data.columns[destination.droppableId];
+
+    if (start === finish) {
+      const newTaskIds = Array.from(start.taskIds);
+      newTaskIds.splice(source.index, 1);
+      newTaskIds.splice(destination.index, 0, draggableId);
+
+      const newColumn = {
+        ...start,
+        taskIds: newTaskIds,
+      };
+
+      const newState = {
+        ...data,
+        columns: {
+          ...data.columns,
+          [newColumn.id]: newColumn,
+        },
+      };
+      await setData(newState);
+      await saveBoard(data.id, {columns: newState.columnOrder})
+      return;
+    }
+
+    // between lists
+    const startTaskIds = Array.from(start.taskIds);
+    startTaskIds.splice(source.index, 1);
+    const newStart = {
+      ...start,
+      taskIds: startTaskIds,
+    };
+
+    const finishTaskIds = Array.from(finish.taskIds);
+    finishTaskIds.splice(destination.index, 0, draggableId);
+    const newFinish = {
+      ...finish,
+      taskIds: finishTaskIds,
+    };
+
+    const newState = {
+      ...data,
+      columns: {
+        ...data.columns,
+        [newStart.id]: newStart,
+        [newFinish.id]: newFinish,
+      },
+    };
+    setData(newState);
+  };
+
+  return (
+    <DragDropContext onDragEnd={onDragEnd}>
+    {/* // Add Left and Right Hover Bars for adding columns */}
+    <div className={classes.addColumnContainer} id="leftNav"> 
+      {data.columns ? (
+        <AddColumnSidebar 
+          data={data}
+          setData={setData}
+          setLoadBoard={setUpdate}
+          boardId={data.id}
+        />
```

Comment:
I only added the column side bars and refactored it to work locally so I can add columns without manually creating them in the backend. Those are good points to keep me in check for because I do seem to break the single-responsibility principle by habit. Thanks @ugur-akin. HEHE. @swathivottigunta and I discussed how we will be moving forward with her contributions and I will likely be deleting my own edits to avoid merge conflicts. =]